src/platforms/f3/platform.c

```diff
@@ -112,9 +118,71 @@ bool platform_nrst_get_val(void)
 	return (gpio_get(NRST_PORT, NRST_PIN)) ? false : true;
 }
 
+
+static void adc_init(void)
+{
+
+	//ADC
+	rcc_periph_clock_enable(RCC_ADC12);
+	//ADC
+	gpio_mode_setup(VTREF_PORT, GPIO_MODE_ANALOG, GPIO_PUPD_NONE, VTREF_PIN);
+	//gpio_mode_setup(GPIOA, GPIO_MODE_ANALOG, GPIO_PUPD_NONE, GPIO1);
+	adc_power_off(ADC1);
+	adc_set_clk_prescale(ADC1, ADC_CCR_CKMODE_DIV2);
+	adc_set_single_conversion_mode(ADC1);
+	adc_disable_external_trigger_regular(ADC1);
+	adc_set_right_aligned(ADC1);
+	/* We want to read the temperature sensor, so we have to enable it. */
+	//adc_enable_temperature_sensor();
+	adc_set_sample_time_on_all_channels(ADC1, ADC_SMPR_SMP_601DOT5CYC);
+	uint8_t channel_array[] = { 1 }; /* ADC1_IN1 (PA0) */
+	adc_calibrate(ADC1);
+	adc_set_regular_sequence(ADC1, 1, channel_array);
+	adc_set_resolution(ADC1, ADC_CFGR1_RES_12_BIT);
+	adc_power_on(ADC1);
+
+	/* Wait for ADC starting up. */
+	int i;
+	for (i = 0; i < 800000; i++)
+		__asm__("nop");
```

Comment:
`i` cannot go negative, so to avoid signed overflow issues and other such unnecessary badness, please use unsigned types here. `size_t` is ideal.

Note that the preferred style for this sort of busy loop is the one used in the native platform for this:
```c
	for (volatile size_t i = 0U; i < 800000U; ++i)
		continue;
```

The `U` suffix on the constants are not optional and makes them unsigned, preventing a signed-unsigned conversion that can drastically change the meaning of the code.